assembly/src/ast/mod.rs

```diff
@@ -100,9 +100,10 @@ fn check_unused_imports(import_info: &ModuleImports) {
     let import_lib_paths = import_info.import_paths();
     let invoked_procs_paths: Vec<&LibraryPath> =
         import_info.invoked_procs().iter().map(|(_id, (_name, path))| path).collect();
+    let reexported_procs_paths = import_info.reexported_procs_paths();
 
     for lib in import_lib_paths {
-        if !invoked_procs_paths.contains(&lib) {
+        if !invoked_procs_paths.contains(&lib) && !reexported_procs_paths.contains(lib) {
             event!(Level::WARN, "unused import: \"{}\"", lib);
         }
     }
```

Comment:
Most of this function could go into `ModuleImports`, right? For example, we could have something like:
```Rust
pub fn hash_unused_imports(&self) -> bool
```
in `ModuleImports`, and then here, we'd check and print the warning.